getline.ts/src/common.ts

```diff
@@ -92,6 +96,72 @@ export class Token extends Address {
         let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
         return token.call<void>('approve', spender.ascii, value);
     }
+
+    /**
+     * Returns this token's symbol.
+     * @returns Token symbol.
+     */
+    public async symbol(): Promise<string> {
+        if (this.symbol_ != undefined) {
+            return this.symbol_;
+        }
+        let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
+        this.symbol_ = await token.call<string>('symbol');
+        return this.symbol_;
+    }
+
+    /**
+     * Returns this token's name.
+     * @returns Token name.
+     */
+    public async name(): Promise<string> {
+        if (this.name_ != undefined) {
+            return this.name_;
+        }
+        let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
+        this.name_ = await token.call<string>('name');
+        return this.name_;
+    }
+
+    /**
+     * Returns this token's decimal places count.
+     * @returns Decimal places of token.
+     */
+    public async decimals(): Promise<number> {
+        if (this.decimals_ != undefined) {
+            return this.decimals_;
+        }
+        let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
+        // ERC20 dictates uint8 as decimals, but we (seemingly as a bug)
```

Comment:
Preffer BigNumber and quick-fix-assert check when setting variable instead of converting to javascript lossy number - fail fast

Reply:
Done.